examples/CMakeLists.txt

```diff
@@ -18,7 +18,7 @@ foreach(file ${files})
     get_filename_component(target ${file} NAME_WE)
 
     # Exclude SHM examples if SHM feature is disabled
-    if(NOT(ZENOHC_BUILD_WITH_SHARED_MEMORY))
+    if(NOT(ZENOHC_BUILD_WITH_SHARED_MEMORY AND (ZENOHC_BUILD_WITH_UNSTABLE_API)))
```

Comment:
the root cmake already throws error is shared memory is enabled without unstable

Reply:
That is not correct:
- SHM feature enables SHM backend (SHM buffer reception and access via non-SHM APIs)
- SHM + Unstable features enable both SHM backend and SHM API.

So, It is a working case to build with SHM and without Unstable. Currently we have a corner case where the whole SHM API is Unstable.